pinot-core/src/main/java/org/apache/pinot/core/query/request/ServerQueryRequest.java

```diff
@@ -107,6 +107,28 @@ public ServerQueryRequest(Server.ServerRequest serverRequest, ServerMetrics serv
     _timerContext = new TimerContext(_queryContext.getTableName(), serverMetrics, queryArrivalTimeMs);
   }
 
+  /**
+   * To be used by Time Series Query Engine.
+   */
+  public ServerQueryRequest(QueryContext queryContext, List<String> segmentsToQuery, Map<String, String> metadata,
+      ServerMetrics serverMetrics) {
+    long queryArrivalTimeMs = System.currentTimeMillis();
+    _queryContext = queryContext;
+
+    // Initialize metadata
+    _requestId = Long.parseLong(metadata.getOrDefault(Request.MetadataKeys.REQUEST_ID, "0"));
+    _brokerId = metadata.getOrDefault(Request.MetadataKeys.BROKER_ID, "unknown");
+    _enableTrace = Boolean.parseBoolean(metadata.getOrDefault(Request.MetadataKeys.ENABLE_TRACE, "false"));
+    _enableStreaming = Boolean.parseBoolean(metadata.getOrDefault(Request.MetadataKeys.ENABLE_STREAMING, "false"));
+    _queryId = QueryIdUtils.getQueryId(_brokerId, _requestId,
+        TableNameBuilder.getTableTypeFromTableName(_queryContext.getTableName()));
+
+    _segmentsToQuery = segmentsToQuery;
+    _optionalSegments = null;
+
+    _timerContext = new TimerContext(_queryContext.getTableName(), serverMetrics, queryArrivalTimeMs);
```

Comment:
We should be able to use existing pinot server metrics to troubleshoot any latency or other issues in time series engine, correct ?

Reply:
We will require more metrics for sure. We will pick it up next month since we need to productionize this in October.

Reply:
Can we add this to Misc task list for tracking?

Reply:
Yup added here: https://github.com/apache/pinot/issues/13957